### DISCUSSION TITLE
[ 題目看嘸 ] LeetCode 1640. Check Array Formation Through Concatenation

### DISCUSSION THREAD

**Discussion by LPenny-github** [EngTW/English-for-Programmers#104] [Q&A] [answered]
原 issue:
#85

LeetCode:
https://leetcode.com/problems/check-array-formation-through-concatenation/

---

題目看不懂。麻煩各位大大 orz 感激不盡

已知部分：
題目會給兩個 arr，一個裝 不重複的整數，另一個裝 不重複的整數片段(?)。
在 片段(?) 裡的也是不重複的整數。

引用 LeetCode:

```
You are given an array of distinct integers arr and an array of integer arrays pieces, 
where the integers in pieces are distinct. 
Your goal is to form arr by concatenating the arrays in pieces in any order. 
However, you are not allowed to reorder the integers in each array pieces[i].

Return true if it is possible to form the array arr from pieces. Otherwise, return false.
```

---

我不知道該用什麼格式和標題， @twy30 大大可以自行修改本篇 😉

**Comment by twy30** [comment, marked as answer]
@LPenny-github 

> 題目會給兩個 arr，一個裝 不重複的整數，另一個裝 不重複的整數片段(?)。
> 在 片段(?) 裡的也是不重複的整數。

是的。

題目的要求是：「如果那些片段 (`pieces`) 可以以某種次序連起來組成 `arr` ，就傳回 `true` ；否則，傳回 `false` 。」

---

可以把這想成「拼圖」。

以 LeetCode 原題的 "Example 4" 來說：

* 拼圖碎片 `pieces` 有 3 塊： `[78]`, `[4,64]`, `[91]`
* 最後想要的樣子 `arr` 是 `[91,4,64,78]`
* 答案為 `true`
  * 因為可以用 `[91]`, `[4,64]`, `[78]` 的方式將 `pieces` 組成 `arr` 的樣子

以 LeetCode 原題的 "Example 3" 來說：

* 拼圖碎片 `pieces` 有 1 塊： `[16,18,49]`
* 最後想要的樣子 `arr` 是 `[49,18,16]`
* 答案為 `false`
  * `pieces` 給我們唯一的片段是 `[16,18,49]`, 怎麼樣也無法組成 `arr` 的樣子
  * 因為題目有明確規定「不可變動片段中的成員的順序」
    * > However, you are not allowed to reorder the integers in each array pieces[i]
    * 也就是不能把 `[16,18,49]` 變成 `[49,18,16]`
    * 在 "Example 4" 裡，我們變動的是片段( `pieces` ) 的順序，也就是將 `[78], [4,64], [91]` 變成 `[91], [4,64], [78]`, 但我們沒有去變動「片段的成員的順序」

**Reply by LPenny-github** [reply]
@twy30 感謝大大 orz
